ObjectiveGit/GTRepository+RemoteOperations.h

```diff
@@ -97,6 +97,40 @@ typedef NS_ENUM(NSInteger, GTFetchPruneOption) {
 /// will point to an error describing what happened).
 - (BOOL)pushBranches:(NSArray<GTBranch *> *)branches toRemote:(GTRemote *)remote withOptions:(nullable NSDictionary *)options error:(NSError **)error progress:(nullable void (^)(unsigned int current, unsigned int total, size_t bytes, BOOL *stop))progressBlock;
 
+/// Push an array of branches to a remote, together with notes (in one push).
+/// Normally, if you use Git notes functionality, to push the notes to remote, you would have to
+/// do two operations: pushBranches followed by pushNotes. This is unrational, so we offer you a handy
+/// shortcut that allows to push the branches together with the notes.
+///
+/// branches      - An array of branches to push. Must not be nil.
+/// remote        - The remote to push to. Must not be nil.
+/// options       - Options applied to the push operation. Can be NULL.
+///                 Recognized options are:
+///                 `GTRepositoryRemoteOptionsCredentialProvider`
+/// referenceName - Reference name for notes, if they should be pushed together with the branches.
+///                 Use +[GTNote defaultReferenceNameWithError:] to push the default note reference.
+///                 Passing NULL here will make notes NOT to be pushed.
+/// error         - The error if one occurred. Can be NULL.
+/// progressBlock - An optional callback for monitoring progress. May be NULL.
+///
+/// Returns YES if the push was successful, NO otherwise (and `error`, if provided,
+/// will point to an error describing what happened).
+- (BOOL)pushBranches:(NSArray<GTBranch *> *)branches toRemote:(GTRemote *)remote withOptions:(nullable NSDictionary *)options withNotesReferenceName:(nullable NSString*)notesReferenceName error:(NSError **)error progress:(nullable void (^)(unsigned int current, unsigned int total, size_t bytes, BOOL *stop))progressBlock;
```

Comment:
I'm not particularly fond of a new method just for pushing notes. Maybe a new option (`GTRepositoryRemoteOptionsNotesReferences`) would be better, taking an array of notes "namespace" ?

Arguably, it stems from the fact that this is the branch-pushing mechanism, and branches don't really exist (everything's just a reference to git), and notes are another "kind" of references.

@pietbrauer Opinion ? I'm not that comfortable with notes, as I've never used them 😟.